src/main/java/org/kuali/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java

```diff
@@ -0,0 +1,312 @@
+package org.kuali.kfs.gl.batch.service.impl;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.coa.businessobject.A21SubAccount;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.businessobject.OffsetDefinition;
+import org.kuali.kfs.gl.GeneralLedgerConstants;
+import org.kuali.kfs.gl.batch.ScrubberStep;
+import org.kuali.kfs.gl.businessobject.OriginEntryFull;
+import org.kuali.kfs.gl.businessobject.OriginEntryInformation;
+import org.kuali.kfs.gl.businessobject.Transaction;
+import org.kuali.kfs.gl.service.ScrubberReportData;
+import org.kuali.kfs.gl.service.impl.StringHelper;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.kfs.sys.Message;
+import org.kuali.kfs.sys.businessobject.SystemOptions;
+import org.kuali.kfs.sys.exception.InvalidFlexibleOffsetException;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+
+/**
+ * Cornell's implementation of ScrubberProcess that extends from the KFS-delivered ScrubberProcessImpl.
+ * 
+ * NOTE: Because ScrubberProcessImpl declares an internal TransactionError class with default (package-private)
+ * visibility, this subclass has to be in the same package as ScrubberProcessImpl in order to use that
+ * internal class. If future versions of KFS update that class with a higher visibility, then move
+ * this subclass into a more appropriate package.
+ */
+public class CuScrubberProcessImpl extends ScrubberProcessImpl {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuScrubberProcessImpl.class);
+
+    private static final int CONTINUATION_ACCOUNT_DEPTH_LIMIT = 10;
+
+    /**
+     * Overridden to update cost share source account entries so that they will post to the
+     * continuation account if the cost share account is closed.
+     * 
+     * @see org.kuali.kfs.gl.batch.service.impl.ScrubberProcessImpl#generateCostShareEntries(
+     *         org.kuali.kfs.gl.businessobject.OriginEntryInformation, org.kuali.kfs.gl.service.ScrubberReportData)
+     */
+    @Override
+    protected TransactionError generateCostShareEntries(OriginEntryInformation scrubbedEntry, ScrubberReportData scrubberReport) {

```

Comment:
This is mostly the same as the method from the superclass, except for the portion that calls a new method for doing the Cost Share and continuation account setup.